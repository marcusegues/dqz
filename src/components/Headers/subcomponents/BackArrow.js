import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import Touchable from 'react-native-platform-touchable';
import { MAIN_RED } from '../../../styles/colors';

const BackArrow = ({ onPress }) => (
  <Touchable onPress={onPress}>
    <Ionicons name="md-arrow-back" size={32} color={MAIN_RED} />
  </Touchable>
);

export default BackArrow;
