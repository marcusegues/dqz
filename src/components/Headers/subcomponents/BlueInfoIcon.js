// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';

export const BlueInfoIcon = () => (
  <Touchable onPress={() => {}}>
    <MaterialCommunityIcons
      name="information"
      size={moderateScale(30)}
      color="#006699"
    />
  </Touchable>
);
