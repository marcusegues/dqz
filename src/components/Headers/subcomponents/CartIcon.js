// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { receiptNotificationBadgeStyle } from '../styles/ReceiptNotificationBadge';
import { MAIN_RED } from '../../../styles/colors';

type CartIconProps = {
  navigation: Navigation,
};

export const CartIcon = ({ navigation }: CartIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'QuestionAnswer' })
    }
  >
    <MaterialIcons
      name="shopping-cart"
      size={moderateScale(30)}
      color={MAIN_RED}
      style={receiptNotificationBadgeStyle.receiptIcon}
    />
  </Touchable>
);
