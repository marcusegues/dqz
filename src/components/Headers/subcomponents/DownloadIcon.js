// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { MAIN_RED } from '../../../styles/colors';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';

type DownloadIconProps = {
  navigation: Navigation,
};

export const DownloadIcon = ({ navigation }: DownloadIconProps) => (
  <Touchable
    onPress={() => {
      navigation.goBack();
    }}
  >
    <MaterialIcons name="get-app" size={moderateScale(28)} color={MAIN_RED} />
  </Touchable>
);
