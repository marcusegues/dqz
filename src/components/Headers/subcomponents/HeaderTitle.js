import React from 'react';

import { Text } from 'react-native';

const HeaderTitle = ({ text }) => (
  <Text
    style={{
      fontSize: 18,
      color: '#1A1A1A',
      fontFamily: 'roboto_regular',
    }}
  >
    {text}
  </Text>
);

export default HeaderTitle;
