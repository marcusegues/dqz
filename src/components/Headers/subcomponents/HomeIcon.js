// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import Entypo from '@expo/vector-icons/Entypo';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { GREY } from '../../../styles/colors';

type HomeIconProps = {
  navigation: Navigation,
};

export const HomeIcon = ({ navigation }: HomeIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'MainMenu' })
    }
  >
    <Entypo name="home" size={moderateScale(27)} color={GREY} />
  </Touchable>
);
