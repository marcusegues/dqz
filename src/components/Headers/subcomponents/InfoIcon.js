import React from 'react';
import Touchable from 'react-native-platform-touchable';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';
import { GREY } from '../../../styles/colors';

const InfoIcon = ({ navigation }) => (
  <Touchable
    onPress={() => {
      navigation.goBack();
    }}
  >
    <MaterialCommunityIcons
      name="information"
      size={moderateScale(23)}
      color={GREY}
    />
  </Touchable>
);

export default InfoIcon;
