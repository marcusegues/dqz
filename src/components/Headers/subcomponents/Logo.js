// @flow
import React from 'react';
// $FlowFixMe
import { Image } from 'react-native';

const logo = require('../../../../assets/images/logo.png');

export const Logo = () => (
  <Image source={logo} resizeMode="contain" style={{ width: 30, height: 30 }} />
);
