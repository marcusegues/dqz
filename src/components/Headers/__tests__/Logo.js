import renderer from 'react-test-renderer';
import React from 'react';
import { Logo } from '../subcomponents/Logo';

describe('BackArrow', () => {
  test('renders according to snapshot', () => {
    const component = renderer.create(<Logo />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
