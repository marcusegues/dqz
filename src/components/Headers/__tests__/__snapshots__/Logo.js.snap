// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Logo renders according to snapshot 1`] = `
<Image
  resizeMode="contain"
  source={1}
  style={
    Object {
      "height": 30,
      "width": 30,
    }
  }
/>
`;
